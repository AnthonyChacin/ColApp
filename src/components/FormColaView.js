import React, { Component } from 'react';
import {
    AppRegistry,
    Platform, Image,
    TextInput,
    Button,
    StyleSheet,
    Text,
    View,
    TouchableOpacity,
    Picker
} from 'react-native';

import DatePicker from 'react-native-datepicker';

import axios from 'axios';

class FormColaView extends React.Component {

    constructor() {
        super();
        this.state = {
            origen: '',
            destino: '',
            tarifa: '',
            banco: '',
            hora: new Date(),
            vehiculo: '',
            cantPasajeros: ''
        }
    }

    updateValue(param, i) {
        switch (i) {
            case 1:
                this.setState({
                    origen: param
                })
                break;
            case 2:
                this.setState({
                    destino: param
                })
                break;
            case 3:
                this.setState({
                    tarifa: param
                })
                break;
            case 4:
                this.setState({
                    banco: param
                })
                break;
            case 5:
                this.setState({
                    hora: param
                })
                break;
            case 6:
                this.setState({
                    cantPasajeros: param
                })
                break;
        }
     }
 
    async submit() {
         try {

          var url = 'https://colapp-is.herokuapp.com/pasajero/pedirCola';
 
          let cola = await axios.post(url, {
            origen: this.state.origen,
            destino: this.state.destino,
            tarifa: this.state.tarifa,
            banco: this.state.banco,
            hora: this.state.hora,
            cantPasajeros: this.state.cantPasajeros,
            vehiculo: this.state.vehiculo,
            estado: "Pedida",
            pasajero: this.props.navigation.getParam('PasajeroId', 'No-Id')
          })

          console.warn(cola)
 
          if (cola.data.success) {
            this.setState({
              origen: '',
              destino: '',
              tarifa: '',
              banco: '',
              hora: new Date(),
              cantPasajeros: '',
              vehiculo: ''
            })
          }
 
         }catch (error) {
            console.warn('Error al pedir la cola')
         }
    }

    static navigationOptions = {
        header: null
    }

    render() {
        return (

            <View style={styles.container}>

                <TextInput
                    placeholder="Origen"
                    style={styles.textInput}
                    editable={true}
                    underlineColorAndroid="transparent"
                    onChangeText={(text) => this.updateValue(text, 1)}
                    value={this.state.origen}

                />
                <TextInput
                    placeholder="Destino"
                    style={styles.textInput}
                    editable={true}
                    underlineColorAndroid="transparent"
                    onChangeText={(text) => this.updateValue(text, 2)}
                    value={this.state.destino}

                />
                <TextInput
                    placeholder="Tarifa"
                    style={styles.textInput}
                    editable={true}
                    underlineColorAndroid="transparent"
                    onChangeText={(text) => this.updateValue(text, 3)}
                    value={this.state.tarifa}
                    keyboardType="numeric"
                />
                <TextInput
                    placeholder="Banco"
                    style={styles.textInput}
                    editable={true}
                    underlineColorAndroid="transparent"
                    onChangeText={(text) => this.updateValue(text, 4)}
                    value={this.state.banco}
                />
                <DatePicker
                    date = {this.state.hora}
                    mode = 'datetime'
                    placeholder = 'seleccione fecha y hora'
                    format = "YYYY-MM-DD hh:mm:ss"
                    minDate = {new Date()}
                    confirmBtnText = "Confirm"
                    cancelBtnText = "Cancel"
                    customStyles = {{
                      dateIcon: {
                        position: 'absolute',
                        left: 0,
                        top: 4,
                        marginLeft: 0
                      },
                      dateInput: {
                        marginLeft: 36
                      }
                    }}
                    onDateChange = {(date) => {this.updateValue(date, 5)}}
                />
                <TextInput
                    placeholder="Cantidad de Pasajeros"
                    style={styles.textInput}
                    editable={true}
                    underlineColorAndroid="transparent"
                    onChangeText={(text) => this.updateValue(text, 6)}
                    value={this.state.cantPasajeros}
                    keyboardType="numeric"
                />
                <Picker
                    selectedValue={this.state.vehiculo}
                    style={styles.Picker}
                    onValueChange={(itemValue, itemIndex) =>
                        this.setState({ vehiculo: itemValue })
                    }>
                    <Picker.Item label="Moto" value="moto" />
                    <Picker.Item label="Carro" value="carro" />
                </Picker>
                <TouchableOpacity
                    style={styles.buttonSubmit}
                    onPress={() => this.submit(1)}
                >
                    <Text style={{ color: "white", fontSize: 20 }}>Solicitar</Text>
                </TouchableOpacity>
            </View>
        );
    }
}

export default FormColaView;

const styles = StyleSheet.create({

    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgb(20,20,20)',

    },
    buttonSubmit: {
        paddingHorizontal: 16,
        paddingTop: 10,
        paddingBottom: 10,
        backgroundColor: "#E6880F",
        alignItems: "center",
        color: "white",
        alignSelf: "center",
        marginTop: 10,
        marginBottom: 10,
        borderRadius: 25,
        width: 300
    },
    button: {
        paddingHorizontal: 16,
        paddingTop: 10,
        paddingBottom: 10,
        backgroundColor: "#E6880F",
        alignItems: "center",
        color: "white",
        alignSelf: "center",
        marginTop: 10,
        marginBottom: 10,
        // borderRadius: 25,
        width: 300
    },
    textInput: {
        alignSelf: 'center',
        paddingHorizontal: 16,
        marginTop: 10,
        marginBottom: 20,
        backgroundColor: "#fff",
        // borderRadius: 25,
        width: 300
    },
    Picker: {
        alignSelf: 'center',
        paddingHorizontal: 16,
        marginTop: 10,
        marginBottom: 20,
        backgroundColor: "#fff",
        borderWidth: 1,
        overflow: 'hidden',
        // borderRadius: 10,
        width: 300
    },
    welcome: {
        fontSize: 25,
        textAlign: 'center',
        margin: 20,
        color: '#E6880F',
        fontFamily: 'Arial'
    }
});
