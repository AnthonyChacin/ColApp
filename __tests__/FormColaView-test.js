/**
 * @format
 */

import 'react-native';

import React from 'react';
import FormColaView from '../src/components/FormColaView';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('Se renderiza correctamente el formulario donde se pedirá la cola', () => {

  	var ColaData = renderer.create(<FormColaView />).toJSON()
	expect(ColaData).toMatchSnapshot();
});

test('verificación del envío de datos del formulario para pedir cola', () => {
	let DataPost = renderer.create(<FormColaView />).getInstance();

	expect(DataPost.submit()).toEqual(true)
})
