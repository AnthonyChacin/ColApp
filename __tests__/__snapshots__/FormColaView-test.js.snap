// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se renderiza correctamente el formulario donde se pedir√° la cola 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(20,20,20)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    placeholder="Origen"
    placeholderTextColor="rgba(20,20,20,0.3)"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "marginBottom": 20,
        "marginTop": 10,
        "paddingHorizontal": 16,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    placeholder="Destino"
    placeholderTextColor="rgba(20,20,20,0.3)"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "marginBottom": 20,
        "marginTop": 10,
        "paddingHorizontal": 16,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    editable={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Tarifa"
    placeholderTextColor="rgba(20,20,20,0.3)"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "marginBottom": 20,
        "marginTop": 10,
        "paddingHorizontal": 16,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    placeholder="Banco"
    placeholderTextColor="rgba(20,20,20,0.3)"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "marginBottom": 20,
        "marginTop": 10,
        "paddingHorizontal": 16,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "width": 142,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#aaa",
              "borderWidth": 1,
              "flex": 1,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "marginLeft": 36,
            },
            false,
            false,
          ]
        }
      >
        <Text
          allowFontScaling={true}
          style={
            Array [
              Object {
                "color": "#333",
              },
              undefined,
            ]
          }
        >
          2019-05-31 10:33:49
        </Text>
      </View>
      <Image
        source={
          Object {
            "testUri": "../../react-native-datepicker/date_icon.png",
          }
        }
        style={
          Array [
            Object {
              "height": 32,
              "marginLeft": 5,
              "marginRight": 5,
              "width": 32,
            },
            Object {
              "left": 0,
              "marginLeft": 0,
              "position": "absolute",
              "top": 4,
            },
          ]
        }
      />
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "portrait-upside-down",
            "landscape",
            "landscape-left",
            "landscape-right",
          ]
        }
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#00000077",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "height": 0,
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  pointerEvents="auto"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderTopColor": "#ccc",
                          "borderTopWidth": 1,
                          "marginTop": 42,
                        },
                        undefined,
                      ]
                    }
                  >
                    <RCTDatePicker
                      date={1559356429970}
                      minimumDate={1559356429970}
                      mode="datetime"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "height": 216,
                        }
                      }
                      timeZoneOffsetInMinutes={null}
                    />
                  </View>
                </View>
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "left": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      Array [
                        Object {
                          "color": "#46cf98",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#666",
                        },
                        undefined,
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "right": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      Array [
                        Object {
                          "color": "#46cf98",
                          "fontSize": 16,
                        },
                        undefined,
                      ]
                    }
                  >
                    Confirm
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    editable={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Cantidad de Pasajeros"
    placeholderTextColor="rgba(20,20,20,0.3)"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "marginBottom": 20,
        "marginTop": 10,
        "paddingHorizontal": 16,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
        "overflow": "hidden",
        "paddingHorizontal": 16,
        "width": 300,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Moto",
            "textColor": undefined,
            "value": "moto",
          },
          Object {
            "label": "Carro",
            "textColor": undefined,
            "value": "carro",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#E6880F",
        "borderRadius": 25,
        "color": "white",
        "marginBottom": 10,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingHorizontal": 16,
        "paddingTop": 10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
    >
      Solicitar
    </Text>
  </View>
</View>
`;
